pulsar-broker/src/main/java/org/apache/pulsar/zookeeper/MaxValueMetadataNodePayloadLenEstimator.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.zookeeper;
+
+import java.util.HashMap;
+import java.util.List;
+import org.apache.pulsar.common.util.StringSplitter;
+import org.apache.pulsar.metadata.api.GetResult;
+import org.apache.pulsar.metadata.api.MetadataNodePayloadLenEstimator;
+
+public class MaxValueMetadataNodePayloadLenEstimator implements MetadataNodePayloadLenEstimator {
+
+    private final HashMap<PathType, Integer> maxLenPerPathType = new HashMap<>();
+    private final HashMap<PathType, Integer> maxLenPerListPathType = new HashMap<>();
```

Comment:
HashMap isn't thread safe. It's better to use ConcurrentHashMap to avoid problems.

Reply:
> It is not important enough to ensure thread safety for an accurate value, so skipped to guarantee the thread safety

In Java, that's not a proper way to handle thread safety. Another thread might never "see" the changes made by another thread. In practice this rarely happens, but it can happen due to compiler optimizations. That's why such patterns should be avoided in general since they lead to non-deterministic and inconsistent behavior.